tensorflow_datasets/text/disco_fuse.py

```diff
@@ -0,0 +1,166 @@
+            "discourse_type": tfds.features.ClassLabel(names=[
+                'PAIR_ANAPHORA',
+                'PAIR_CONN',
+                'PAIR_CONN_ANAPHORA',
+                'PAIR_NONE',
+                'SINGLE_APPOSITION',
+                'SINGLE_CATAPHORA',
+                'SINGLE_CONN_INNER',
+                'SINGLE_CONN_INNER_ANAPHORA',
+                'SINGLE_CONN_START',
+                'SINGLE_RELATIVE',
+                'SINGLE_S_COORD',
+                'SINGLE_S_COORD_ANAPHORA',
+                'SINGLE_VP_COORD']),
+            "connective_string": tfds.features.Text(),
+            "has_coref_type_pronoun": tfds.features.ClassLabel(
+                num_classes=2),
+            "has_coref_type_nominal": tfds.features.ClassLabel(
+                num_classes=2)
+        }),
+        supervised_keys=None,
+        urls=['https://github.com/google-research-datasets/discofuse'],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    url = _URLS[self.builder_config.name]
+    directory = dl_manager.download_and_extract(url)
+    directory = os.path.join(directory, 'discofuse_v1')
+    directory = os.path.join(directory, self.builder_config.name)
+    train = os.path.join(directory, "train.tsv")
+    dev = os.path.join(directory, "dev.tsv")
+    test = os.path.join(directory, "test.tsv")
+    train_balanced = os.path.join(directory, "train_balanced.tsv")
+    dev_balanced = os.path.join(directory, "dev_balanced.tsv")
+    test_balanced = os.path.join(directory, "test_balanced.tsv")
+
+    return [
+        tfds.core.SplitGenerator(
+            name="train",
+            num_shards=2,
+            gen_kwargs={"directory": train},
+        ),
+        tfds.core.SplitGenerator(
+            name="dev",
+            num_shards=1,
+            gen_kwargs={"directory": dev},
+        ),
+        tfds.core.SplitGenerator(
+            name="test",
+            num_shards=1,
+            gen_kwargs={"directory": test},
+        ),
+        tfds.core.SplitGenerator(
+            name="train_balanced",
+            num_shards=1,
+            gen_kwargs={"directory": train_balanced},
+        ),
+        tfds.core.SplitGenerator(
+            name="dev_balanced",
+            num_shards=1,
+            gen_kwargs={"directory": dev_balanced},
+        ),
+        tfds.core.SplitGenerator(
+            name="test_balanced",
+            num_shards=1,
+            gen_kwargs={"directory": test_balanced},
+        ),
+    ]
+
+  def _generate_examples(self, directory):
+    """Yields examples."""
+    with tf.io.gfile.GFile(directory) as tsvfile:
+      reader = csv.DictReader(tsvfile, dialect='excel-tab')
+      for row in reader:
+        row['has_coref_type_pronoun'] = int(
+            float(row['has_coref_type_pronoun']))
+        row['has_coref_type_nominal'] = int(
+            float(row['has_coref_type_nominal']))
+        yield row
```

Comment:
S3 is enabled - please yield a tuple (id, content) -- see other datasets for example.